Cython/Compiler/ModuleNode.py

```diff
@@ -300,17 +300,9 @@ def h_entries(entries, api=0, pxd=0):
             h_code_main.putln("/* WARNING: the interface of the module init function changed in CPython 3.5. */")
             h_code_main.putln("/* It now returns a PyModuleDef instance instead of a PyModule instance. */")
             h_code_main.putln("")
-            h_code_main.putln("#if PY_MAJOR_VERSION < 3")
-            if env.module_name.isascii():
-                py2_mod_name = env.module_name
-            else:
-                py2_mod_name = env.module_name.encode("ascii", errors="ignore").decode("utf-8")
-                h_code_main.putln('#error "Unicode module names are not supported in Python 2";')
-            h_code_main.putln("PyMODINIT_FUNC init%s(void);" % py2_mod_name)
-            h_code_main.putln("#else")
             py3_mod_func_name = self.mod_init_func_cname('PyInit', env)
-            warning_string = EncodedString('Use PyImport_AppendInittab("%s", %s) instead of calling %s directly.' % (
-                py2_mod_name, py3_mod_func_name, py3_mod_func_name))
+            warning_string = EncodedString('Use PyImport_AppendInittab(%s, %s) instead of calling %s directly.' % (
+                env.module_name.as_c_string_literal(), py3_mod_func_name, py3_mod_func_name))
```

Comment:
`as_c_string_literal()` does a lot more than we want here, e.g. escaping and string splitting.  It's fine that this is done for the deprecation warning below, but it seems weird to do it in a comment.

Was there anything wrong with the previous implementation?